lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -7,6 +7,8 @@ class FileThreadPool
         DEFAULT_CONCURRENCY = 2
         MAX_CONCURRENCY = 2
 
+        Thread.abort_on_exception = true
```

Comment:
Do you have any kind of reproduction case that can give us some confidence this will help? [Thread#join](http://ruby-doc.org/core-2.2.0/Thread.html#method-i-join) should raise any exceptions when this isn't set. Do we have reason to think `#join` is not working as intended in JRuby/this engine?